src/Core/src/FileStream.cpp

```diff
@@ -0,0 +1,113 @@
+#include "FileStream.h"
+#include <stdexcept>
+
+namespace OpenLoco
+{
+
+    FileStream::FileStream(const std::filesystem::path path, StreamMode mode)
+    {
+        if (mode == StreamMode::write)
+        {
+            _fstream.open(path, std::ios::out | std::ios::binary);
+            if (!_fstream.is_open())
+            {
+                throw std::runtime_error("Failed to open '" + path.u8string() + "' for writing");
+            }
+            _reading = true;
```

Comment:
i know this is an existing mistake but we don't ever set the read ios flag so we shouldn't say its okay for reading.

Reply:
This is quite weird anyway, I'll refactor it.